src/bin/ion/commands/beta/generate/utils.rs

```diff
@@ -20,49 +20,144 @@ pub struct Import {
     pub(crate) type_name: String,
 }
 
-/// Represent the programming language for code generation.
-#[derive(Debug, Clone, PartialEq)]
-pub enum Language {
-    Rust,
-    Java,
+pub trait Language {
+    /// Provides a file extension based on programming language
+    fn file_extension() -> String;
+
+    /// Returns string representation of programming language
+    fn string_value() -> String;
+
+    /// Provides file name based on programming language standards
+    fn file_name(name: &str) -> String;
```

Comment:
What is the input `name`? An ISL type name? We should spell it out in the doc comment.

Reply:
This method basically converts the given `name` into a file name for generated code based on programming language.
```rust
/// Provides generated code's file name for given `name` based on programming language standards
/// e.g.
///     In Rust, this will return a string casing `name` to [Case::Snake].
///     In Java, this will return a string casing `name` to  [Case::UpperCamel]
```

Reply:
If the input string is always, say, a target language type name, it would be good to communicate that expectation in the method name and/or the doc comment:
```rust
fn file_name_for_type(name: &str) -> String;
```